### ISSUE TITLE
False positives of UnusedPrivateMember in loops

### ISSUE THREAD

**Issue by sergeys-opera** [detekt/detekt#3566]
Sometimes the loop index doesn't have to be used inside the loop, for example, when you write a retry logic:
```
        for (i in 0 until MAX_ATTEMPTS) {
            try {
                return allocateBitmap(size)
            } catch (ignored: OutOfMemoryError) {
                Log.w(TAG, "OOM with size: x=" + size.x + ", y=" + size.y)
            }
            size.x /= 2
            size.y /= 2
        }
```
Detekt reports `i` as being unused private member but it's actually used _internally_ by the loop itself.
However, in many cases such check would be beneficial as usually you want to use the loop index in the loop code block.
I wish there was `_` for the index variable in loops in Kotlin.

## Expected Behavior
No `UnusedPrivateMember` rule violation.

## Observed Behavior
`UnusedPrivateMember` rule violation.

## Steps to Reproduce
See a code snippet above

## Context
I believe there is no way to workaround it - either the check should be disabled for every loop or this false-positive will pop-up from time to time.

## Your Environment
* Version of detekt used: 1.16.0
* Version of Gradle used (if applicable): 6.8

**Comment by marschwar** [comment]
Apart from catching an `OutOfMemoryError` most likely not being such a great idea, you could just do

```
repeat (MAX_ATTEMPTS) {
    ...
}
```

**Comment by sergeys-opera** [comment]
@marschwar great, thanks. I didn't know about `repeat`. I don't think you need to do anything on your side.

**Comment by rockwotj** [comment]
The `repeat` doesn't work well if you're using break or continue.
